### PR TITLE
#1976: adjust behavior of pipeline and defer in mapargs

### PR DESCRIPTION
**Pull request by twschiller**
NOTE: this is a PR against feature/1976-doc-builder-pipeline

What does this PR do?
- Changes the behavior of mapArgs to keep the defer and pipeline expressions. (Previously is returned the value). We have to keep the expressions around so that a block that receives the value knows what kind of value it is. (In the pipeline case, it would just receive an array)
- Removes the mapArgs step from BlockElement. This was breaking the use of outputKeys in the embedded pipeline. (The reducePipeline call in runRendererPipeline already handles calling mapArgs.)

### CODE REVIEW

File: src/components/documentBuilder/render/BlockElement.tsx [hunk 1]
```diff
@@ -43,25 +43,16 @@ const BlockElement: React.FC<BlockElementProps> = ({ pipeline }) => {
...
-
-    const resolvedPipeline = (await runMapArgs(
-      target,
-      // TODO: pass runtime version via DocumentContext instead of hardcoding it
-      {
-        config: pipeline,
-        context: ctxt,
-        options: apiVersionOptions("v3"),
-      }
-    )) as BlockPipeline;
```
Comment (BALEHOK) [inline src/components/documentBuilder/render/BlockElement.tsx:56]: This part is required to resolve List args in the Pipeline config.

**Review by BALEHOK — CHANGES_REQUESTED** [review]
Need to take a closer look and fix the rendering of the pipeline (for me it fails to render in the side panel).

File: src/components/documentBuilder/render/BlockElement.tsx (hunk 1 above)
Comment (BALEHOK) [reply src/components/documentBuilder/render/BlockElement.tsx:56]: I.e. when the Block element is added as a child of a List.

Comment (twschiller) [reply src/components/documentBuilder/render/BlockElement.tsx:56]: @BALEHOK The way to do that is to enrich the inputs passed to the pipeline in the context. E.g., you'd add `@element` to the context

**Review by BALEHOK — APPROVED** [review]

**Pull request merged**